### PR TITLE
bzr: Fix Python interpreter for Linuxbrew

### PR DESCRIPTION
**Pull request by bukzor**

### CODE REVIEW

File: Library/Formula/bazaar.rb [hunk 1]
```diff
@@ -20,11 +20,8 @@ def install
     system "make", "man1/bzr.1"
     man1.install "man1/bzr.1"
 
-    # Put system Python first in path
-    ENV.prepend_path "PATH", "/System/Library/Frameworks/Python.framework/Versions/Current/bin"
-
     system "make"
-    inreplace "bzr", "#! /usr/bin/env python", "#!/usr/bin/python"
+    inreplace "bzr", "#! /usr/bin/env python", "#!#{which "python"}"
```
Comment (sjackman) [inline Library/Formula/bazaar.rb]: Use instead
```ruby
inreplace "bzr", "#! /usr/bin/env python", "#!/usr/bin/python" if OS.mac?
```

**Comment by sjackman** [comment]
The preferred commit log message is `bzr: Fix for Linuxbrew`
Please squash to a single commit after this fixes are made (that is, use `git commit --amend`).

File: Library/Formula/bazaar.rb (hunk 1 above)
Comment (bukzor) [inline Library/Formula/bazaar.rb]: We don't want `/usr/bin/env python` in the case of linux either, for the same reason that it's not wanted in machomebrew; it's too often that a entirely different python is on the path than the one we compiled with. We should pin it down to the one that's on the path during build.

**Comment by bukzor** [comment]
I believe the travis failure wasn't due to my patch.

**Comment by bukzor** [comment]
This has failed four times in a row. Is master branch broken?
I can't see how my patch would have caused that error.

File: Library/Formula/bazaar.rb (hunk 1 above)
Comment (sjackman) [inline Library/Formula/bazaar.rb]: My preference is to trust the user's `PATH`. As a user if I upgrade my Python distribution for example with `brew install python`, I'd prefer that installed packages started using it.

**Comment by sjackman** [comment]
Yes, `master` appears broken. The gem `tins-1.6.0` appears to have gone AWOL.
```
Could not find tins-1.6.0 in any of the sources (Bundler::GemNotFound)
	from /home/travis/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/gems/1.9.1/gems/bundler-1.7.6/lib/bundler/spec_set.rb:85:in `map!'
```

File: Library/Formula/bazaar.rb [hunk 2]
```diff
@@ -20,11 +20,8 @@ def install
     system "make", "man1/bzr.1"
     man1.install "man1/bzr.1"
 
-    # Put system Python first in path
-    ENV.prepend_path "PATH", "/System/Library/Frameworks/Python.framework/Versions/Current/bin"
```
Comment (sjackman) [inline Library/Formula/bazaar.rb]: Use instead
```ruby
ENV.prepend_path "PATH", "/System/Library/Frameworks/Python.framework/Versions/Current/bin" if OS.mac?
```

Comment (sjackman) [inline Library/Formula/bazaar.rb]: Or rather leave this line untouched as it causes no harm on Linux.

**Comment by sjackman** [comment]
I've merged
```ruby
inreplace "bzr", "#! /usr/bin/env python", "#!/usr/bin/python" if OS.mac?
```
Thanks for your contribution, Buck.

**Comment by bukzor** [comment]
@sjackman This still results in breakage, since the python on the $PATH during compilation is (sometimes, and in my case) `$BREW/bin/python`.  `#{which}` would give the correct value.

**Comment by sjackman** [comment]
If the `python` formula is installed, wouldn't that be the desired Python interpreter to use, that is `$BREW/bin/python`? Which Python interpreter do you want to use?

**Comment by bukzor** [comment]
Ah, I read it wrong, sorry. You've set it up to be find python using the $PATH under linux. While that fixes my issue (I believe) it's less reliable to pinning to `#{which python}` during the build, since depending on the exact value of $PATH an incompatible python will be used. This is why homebrew pins the shebang; it just happens to be that /usr/bin/python is invariantly the python they compile with.

**Comment by sjackman** [comment]
I understand your and Homebrew's stance, but I prefer not to pin the shebang and to trust the user's `PATH`.

**Pull request closed without merging**